src/Microsoft.OData.Client/DataServiceClientFormat.cs

```diff
@@ -274,5 +300,47 @@ private static string ChooseMediaType(bool hasSelectQueryOption)
 
             return MimeApplicationJsonODataLight;
         }
+
+        /// <summary>
+        /// Loads the metadata and converts it into an EdmModel that is then used by a dataservice context
+        /// This allows the user to use the DataServiceContext directly without having to manually pass an IEdmModel in the Format
+        /// </summary>
+        /// <returns>A service model to be used in format tracking</returns>
+        internal IEdmModel LoadServiceModelUsingNetwork()
+        {
+            HttpWebRequestMessage httpRequest;
+            //internal method
+            if (InjectMetadataHttpNetworkRequest != null)
+            {
+                httpRequest = InjectMetadataHttpNetworkRequest();
+            }
+            else
+            {
+                DataServiceClientRequestMessageArgs args = new DataServiceClientRequestMessageArgs(
+                    "GET",
+                    context.GetMetadataUri(),
+                    context.UseDefaultCredentials,
+                    context.UsePostTunneling,
+                    new Dictionary<string, string>());
+
+                httpRequest = new HttpWebRequestMessage(args);
+            }
+#if !PORTABLELIB
+            IODataResponseMessage response = httpRequest.GetResponse();
+
+#endif
+#if PORTABLELIB
+            Task<IODataResponseMessage> asyncResponse =
+                Task<IODataResponseMessage>.Factory.FromAsync(httpRequest.BeginGetResponse, httpRequest.EndGetResponse,
+                    httpRequest);
+            IODataResponseMessage response = asyncResponse.GetAwaiter().GetResult();
+#endif
+
+            StreamReader streamReader = new StreamReader(response.GetStream());
+            string xmlResponse = streamReader.ReadToEnd();
+            XmlReader xmlReader = XmlReader.Create(new StringReader(xmlResponse));

```

Comment:
Would be a good idea to wrap these 3 readers in a using statement so that they are properly disposed